src/spec/example.cr

```diff
@@ -11,37 +11,40 @@ module Spec
     # :nodoc:
     def initialize(@parent : Context, @description : String,
                    @file : String, @line : Int32, @end_line : Int32,
-                   @focus : Bool, tags,
+                   @focus : Bool, @async : Bool, tags,
                    @block : (->) | Nil)
       initialize_tags(tags)
     end
 
     # :nodoc:
     def run
-      Spec.root_context.check_nesting_spec(file, line) do
```

Comment:
Is it fine to remove this check?

Reply:
It’s actually required. It checks to see that no two `it` blocks are running at the same time, but that’s literally the point of this PR. 🙂 The check to see whether someone defined an `it` block inside another `it` block was changed in this PR to [using a `running` state](https://github.com/crystal-lang/crystal/pull/8944/files#diff-33cbbaaa190121d6f19970c014d02620R206) and raising if you try to add another example while in that state.